### ISSUE TITLE
`--report-unused-disable-directives`: false "Unused eslint-disable directive" for enabled type-aware rules when running without `--type-aware`

### ISSUE THREAD

**Issue by ahmedelgabri** [oxc-project/oxc#26281]
**oxlint version:** 1.81.0 (latest) and 1.79.0, with `oxlint-tsgolint` 7.0.2001, macOS arm64

When a type-aware rule (e.g. `typescript/require-await`) is enabled in `.oxlintrc.json` and a run does **not** pass `--type-aware`, the rule is silently skipped — but `--report-unused-disable-directives` still judges disable directives that reference it and reports them as unused. The directive is demonstrably load-bearing: running the same setup **with** `--type-aware` produces no unused-directive warning, and deleting the flagged token turns the type-aware run red.

## Repro

`.oxlintrc.json`:

```json
{
	"rules": {
		"no-unused-vars": "error",
		"typescript/require-await": "error"
	}
}
```

`tsconfig.json`:

```json
{
	"compilerOptions": {
		"noEmit": true,
		"strict": true
	},
	"include": ["handler.ts"]
}
```

`handler.ts`:

```ts
// eslint-disable-next-line @typescript-eslint/no-unused-vars, @typescript-eslint/require-await
async function buildRegistry(config: object, handler: () => void) {
	const registry: {entries: unknown[]; onReady?: () => void} = {entries: []}
	registry.onReady = handler
	return registry
}
```

Setup: `npm i oxlint@1.81.0 oxlint-tsgolint`

### 1. Plain run — false positive

```
$ npx oxlint --report-unused-disable-directives handler.ts
handler.ts:1:64: warning: Unused eslint-disable directive (no problems were reported from @typescript-eslint/require-await).
```

Note the same directive is actively suppressing a live `no-unused-vars` finding in this very invocation (see run 3), so the comment cannot be deleted even in non-type-aware runs.

### 2. Same command with `--type-aware` — no warning

```
$ npx oxlint --type-aware --report-unused-disable-directives handler.ts
(no output, exit 0)
```

The directive suppresses a real `typescript(require-await)` diagnostic here, proving run 1's verdict wrong.

### 3. Directive deleted — lint goes red

```
$ npx oxlint --type-aware --report-unused-disable-directives handler.ts   # line 1 removed
handler.ts:1:16: error eslint(no-unused-vars): Function 'buildRegistry' is declared but never used. help: Consider removing this declaration.
handler.ts:1:1: error typescript(require-await): Function has no 'await' expression.
```

### 4. Only the flagged token removed — still red

Removing just `@typescript-eslint/require-await` from the comment (the exact edit the warning suggests):

```
$ npx oxlint --type-aware --report-unused-disable-directives handler.ts
handler.ts:2:1: error typescript(require-await): Function has no 'await' expression.
```

## Expected

No unused-directive warning in run 1. A directive referencing a rule that is enabled in config but was not executed in this run (type-aware rule without `--type-aware` / without tsgolint) cannot be known to be unused; reporting should be limited to rules that actually ran.

## Actual

The skipped rule is treated as "produced no problems" and the directive is flagged. Acting on the warning breaks the type-aware lint run (and, for mixed directives like the one above, the surrounding comment also protects live non-type-aware findings).

## Notes

- File-level `/* eslint-disable @typescript-eslint/no-unused-vars, @typescript-eslint/require-await */` blocks are affected identically (warning at the rule token's column).
- Same result with a `.js` file under `allowJs`/`checkJs`.
- This is the unused-directive-side counterpart of #25614 (type-aware rules accepted in config but silently skipped without `--type-aware`): the silent skip additionally feeds `--report-unused-disable-directives` false positives.
- In split CI setups (fast non-type-aware lane + separate `--type-aware` lane over the same files), the non-type-aware lane permanently flags every directive the type-aware lane depends on.

**Comment by wagenet** [comment]
Cross-linking a second, distinct path to the same symptom, in case it's useful when fixing directive accounting: a false "Unused eslint-disable directive" also occurs *with* `--type-aware`, when the diagnostic's node spans multiple lines.

There the rule does run, and the directive is both ignored and reported unused in the same run, because suppression is keyed to a different line than the one the diagnostic is reported at. Details and a minimal repro: oxc-project/tsgolint#769 (comment).

Different cause from this issue — that one is about the rule being skipped entirely — but both surface as `--report-unused-disable-directives` telling you to delete a load-bearing comment, so a fix for one may not cover the other.

---

Written by Claude (Claude Code) on @wagenet's behalf, disclosed per Oxc's AI usage policy.
